metadata-ingestion/src/datahub/api/entities/dataproduct/dataproduct.py

```diff
@@ -0,0 +1,525 @@
+            print(generation_config)
+
+        mcp = MetadataChangeProposalWrapper(
+            entityUrn=self.urn,
+            aspect=DataProductPropertiesClass(
+                name=self.display_name,
+                description=self.description,
+                assets=[
+                    DataProductAssociationClass(
+                        destinationUrn=asset,
+                        created=self._mint_auditstamp("yaml"),
+                    )
+                    for asset in self.assets
+                ],
+                customProperties=self.properties if self.properties else None,
+                externalUrl=self.external_url if self.external_url else None,
+            ),
+        )
+        yield mcp
+
+        mcp = MetadataChangeProposalWrapper(
+            entityUrn=self.urn,
+            aspect=DomainsClass(
+                domains=[builder.make_domain_urn(self._resolved_domain_urn)]
+            ),
+        )
+        yield mcp
+
+        if self.tags:
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=self.urn,
+                aspect=GlobalTagsClass(
+                    tags=[
+                        TagAssociationClass(tag=builder.make_tag_urn(tag))
+                        for tag in self.tags
+                    ]
+                ),
+            )
+            yield mcp
+
+        if self.terms:
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=self.urn,
+                aspect=GlossaryTermsClass(
+                    terms=[
+                        GlossaryTermAssociationClass(urn=builder.make_term_urn(term))
+                        for term in self.terms
+                    ],
+                    auditStamp=self._mint_auditstamp("yaml"),
+                ),
+            )
+            yield mcp
+
+        if self.owners:
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=self.urn,
+                aspect=OwnershipClass(
+                    owners=[self._mint_owner(o) for o in self.owners]
+                ),
+            )
+            yield mcp
+
+        # Finally emit status
+        yield MetadataChangeProposalWrapper(
+            entityUrn=self.urn, aspect=StatusClass(removed=False)
+        )
+
+    def emit(
+        self,
+        emitter: Union[DatahubRestEmitter, "DatahubKafkaEmitter"],
+        callback: Optional[Callable[[Exception, str], None]] = None,
+    ) -> None:
+        """
+        Emit the DataProduct entity to Datahub
+
+        :param emitter: Datahub Emitter to emit the event
+        :param callback: The callback method for KafkaEmitter if it is used
+        """
+        for mcp in self.generate_mcp():
+            emitter.emit(mcp, callback)
+
+    @classmethod
+    def from_yaml(
+        cls,
+        file: Path,
+        graph: DataHubGraph,
+    ) -> "DataProduct":
+        with open(file) as fp:
+            yaml = YAML(typ="rt")  # default, if not specfied, is 'rt' (round-trip)
```

Comment:
here using ruamel.yaml doesn't preserve anything over just using normal yaml load

Reply:
made a change to preserve this original dictionary, so that it is used on the way back out in case of a diff.